### PR TITLE
Add docs for payment webhooks

### PR DESCRIPTION
**Pull request by maarcingebala**
Docs for https://github.com/mirumee/saleor/pull/7044.

### CODE REVIEW

**Review by patrys — APPROVED** [review]

File: docs/developer/extending/apps/key-concepts.mdx [hunk 1]
```diff
@@ -145,6 +146,180 @@ FIFO queues (ending in `.fifo`) must be configured to have `ContentBasedDeduplic
...
+Each item on the list should have the following fields:
+- `id` - ID of a payment method; must be unique within your app.
+- `name` - the name of the payment method.
+- `currencies` - list of currencies supported by this payment method.
+- `config` - optional configuration, such as additional payment data that may be needed to proceed with the payment in the frontend.
+
+
+#### Register webhooks for payment actions
+
+To subscribe to particular payment events (such as capture or authorization), you can have multiple webhooks with separate URLs for each action, or a single webhook that handles all events (each webhook has `X-Saleor-Event` header which contains the type of event).
```
Comment (karolkielecki) [inline docs/developer/extending/apps/key-concepts.mdx:252]:
```suggestion
To subscribe to particular payment events (such as capture or authorization), you can have multiple webhooks with separate URLs for each action or a single webhook that handles all events (each webhook has an `X-Saleor-Event` header that contains the type of event).
```

**Review by karolkielecki — CHANGES_REQUESTED** [review]

**Review by karolkielecki — APPROVED** [review]

File: docs/developer/extending/apps/key-concepts.mdx [hunk 2]
```diff
@@ -145,6 +146,193 @@ FIFO queues (ending in `.fifo`) must be configured to have `ContentBasedDeduplic
...
+
+- `action_required` - (boolean) determines if additional action (confirmation) is required.
+- `action_required_data` - (object) additional data that may be needed to confirm the payment, for example, URL to redirect the user to the confirmation page in the frontend.
+- `amount` - (number) transaction amount; use this field only if the charged amount is different than checkout's total.
+- `customer_id` - (string) gateway's customer ID.
+- `error` - (string) error message; if provided, the transaction will be marked as failed in Saleor and payment will have to be retried. Use this field if you need to return any errors.
+- `payment_method` - (object) information about payment method chosen by the user.
+- `psp_reference` - (string) reference which can be used to find a payment object in Saleor, once the order is created.
+- `transaction_id` - (string) gateway's transaction ID
+- `transaction_already_processed` - (boolean) use this field to mark a transaction as already processed in your gateway
```
Comment (krzysztofwolski) [inline docs/developer/extending/apps/key-concepts.mdx:287]:
```suggestion
- `transaction_id` - (string) gateway's transaction ID.
- `transaction_already_processed` - (boolean) use this field to mark a transaction as already processed in your gateway.
```

File: docs/developer/extending/apps/key-concepts.mdx [hunk 3]
```diff
@@ -145,6 +146,193 @@ FIFO queues (ending in `.fifo`) must be configured to have `ContentBasedDeduplic
...
+     "name":"John Doe",
+     "type":"Credit card",
+   },
+   "transaction_id":"transaction-1234"
+}
+```
+
+#### Use payment webhook's payment method in checkout
+
+If payment webhooks are configured properly, you can use their payment methods in checkout. To do this, you need to [list available payment methods for your checkout or shop](/developer/extending/apps/key-concepts.mdx#listing-all-available-payment-gateways). Each payment method from a payment webhook has an ID of this format: `app:<app-database-id>:<id-returned-by-webhook>`
```
Comment (krzysztofwolski) [inline docs/developer/extending/apps/key-concepts.mdx:312]: Enabling webhook is required? AFAIR by default new webhooks created in dashboard are not enabled

File: docs/developer/extending/apps/sample-webhook-payloads.mdx [hunk 4]
```diff
@@ -1973,3 +1973,122 @@ Each webhook payload has `notify_event` field which defines the type of the noti
...
+</details>
+
+### Payment actions payload
+
+<details><summary>Expand example</summary>
+
+```json
+{
+   "amount":"29.260",
+	 "gateway":"app:1:credit-card"
```
Comment (krzysztofwolski) [inline docs/developer/extending/apps/sample-webhook-payloads.mdx:2054]:
```suggestion
	 "gateway":"app:1:credit-card",
```

File: docs/developer/extending/apps/sample-webhook-payloads.mdx [hunk 5]
```diff
@@ -1973,3 +1973,122 @@ Each webhook payload has `notify_event` field which defines the type of the noti
...
+   },
+   "payment_id":33,
+   "graphql_payment_id":"UGF5bWVudDozMw==",
+   "order_id":null,
+   "customer_ip_address":"127.0.0.1",
+   "customer_email":"john.doe@example.com",
+   "token":"FULLY_CHARGED",
+   "customer_id":null,
+   "reuse_source":false,
+	 "payment_method":"credit-card"
```
Comment (krzysztofwolski) [inline docs/developer/extending/apps/sample-webhook-payloads.mdx:2090]:
```suggestion
	"payment_method":"credit-card",
```

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

File: docs/developer/extending/apps/key-concepts.mdx (hunk 3 above)
Comment (maarcingebala) [reply docs/developer/extending/apps/key-concepts.mdx:312]: Yes, webhook needs to be active.

**Comment by krzysztofwolski** [comment]
@maarcingebala there seem to be some conflicts on the branch, could you rebase?

**Comment by maarcingebala** [comment]
@krzysztofwolski Done

**Review by karolkielecki — APPROVED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Pull request merged**